pkg/network/event_common.go

```diff
@@ -260,27 +260,14 @@ func (c ConnectionStats) IsExpired(now uint64, timeout uint64) bool {
 	return c.LastUpdateEpoch+timeout <= now
 }
 
-// ByteKey returns a unique key for this connection represented as a byte array
+// ByteKey returns a unique key for this connection represented as a byte slice
 // It's as following:
 //
 //     4B      2B      2B     .5B     .5B      4/16B        4/16B   = 17/41B
 //    32b     16b     16b      4b      4b     32/128b      32/128b
 // |  PID  | SPORT | DPORT | Family | Type |  SrcAddr  |  DestAddr
-func (c ConnectionStats) ByteKey(buf []byte) ([]byte, error) {
-	n := 0
-	// Byte-packing to improve creation speed
-	// PID (32 bits) + SPort (16 bits) + DPort (16 bits) = 64 bits
-	p0 := uint64(c.Pid)<<32 | uint64(c.SPort)<<16 | uint64(c.DPort)
-	binary.LittleEndian.PutUint64(buf[0:], p0)
-	n += 8
-
-	// Family (4 bits) + Type (4 bits) = 8 bits
-	buf[n] = uint8(c.Family)<<4 | uint8(c.Type)
-	n++
-
-	n += c.Source.WriteTo(buf[n:]) // 4 or 16 bytes
-	n += c.Dest.WriteTo(buf[n:])   // 4 or 16 bytes
-	return buf[:n], nil
+func (c ConnectionStats) ByteKey(buf []byte) []byte {
```

Comment:
nit: I would prefer something like another function `NATByteKey` or `ByteKeyWithNAT` (exported or not).